addons/account/models/account_account.py

```diff
@@ -372,6 +373,19 @@ def _inverse_code(self):
         }
         self.env['ir.property'].with_company(self.env.company.root_id).sudo()._set_multi('code', 'account.account', values)
 
+    @api.depends_context('company')
+    @api.depends('code')
+    def _compute_placeholder_code(self):
+        self.placeholder_code = False
+        for record in self:
+            if record.code:
+                record.placeholder_code = record.code
+            elif authorized_companies := (record.company_ids & self.env['res.company'].browse(self.env.user._get_company_ids())):
+                for company in authorized_companies:
+                    if code := record.with_company(company).code:
```

Comment:
Why do you even need to loop?
Doesn't a company that is set in `record.company_ids` always have a code set on the account?

Reply:
It's not necessarily the case in an onchange :smile: 
e.g. if you create a new account

Reply:
You're right - either it's a new record, in which case it will not have a code for any company, and otherwise, it will necessarily have a code in all the companies of the `company_ids`. So indeed, no need to loop - I've changed it.